backend/backend.go

```diff
@@ -122,3 +122,38 @@ func UnmarshalHistoryEvent(bytes []byte) (*HistoryEvent, error) {
 	}
 	return e, nil
 }
+
+func GetSubOrchestrationInstances(ctx context.Context, be Backend, iid api.InstanceID, isPurge bool) ([]api.InstanceID, error) {
+	owi := &OrchestrationWorkItem{
+		InstanceID: iid,
+	}
+	state, err := be.GetOrchestrationRuntimeState(ctx, owi)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch orchestration state for '%s': %w", iid, err)
+	}
+	if isPurge {
+		if len(state.NewEvents())+len(state.oldEvents) == 0 {
+			// If there are no events, the orchestration instance doesn't exist
+			return nil, api.ErrInstanceNotFound
+		}
+		if !state.IsCompleted() {
+			// Orchestration must be completed before purging its state
+			return nil, api.ErrNotCompleted
+		}
+	}
+	oldEvents := state.OldEvents()
+	newEvents := state.NewEvents()
+
+	subOrchestrationInstances := make([]api.InstanceID, 0, 10)
```

Comment:
Why the magic number 10? If anything, shouldn't it be `len(oldEvents)+len(newEvents)`?

Reply:
Done